src/intercom_clj/users.clj

```diff
@@ -41,30 +41,40 @@
   Uses the scroll API
   docs: https://developers.intercom.com/intercom-api-reference/v1.0/reference#iterating-over-all-users"
   ([] (all nil))
-  ([page-limit]
-   (loop [pages-left page-limit
-          acc []
+  ([page-limit & {:keys [retries]
+                  :or   {retries 0}}]
+   (loop [pages-left   page-limit
+          acc          []
+          retries-left retries
           scroll-param nil]
      (info "INTERCOM_SCROLL_FETCHED" (count acc) "users")
      (if (or (nil? pages-left) (> pages-left 0))
        (let [{:keys [body status]} (scroll scroll-param)]
-         (if (= 200 status)
-           (let [users (:users body)]
-             (if (empty? users)
-               acc
-               (recur (when pages-left (dec pages-left))
-                      (vec (concat acc users))
-                      (:scroll_param body))))
+         (condp = status
+           200 (let [users (:users body)]
+                 (if (empty? users)
+                   acc
+                   (recur (when pages-left (dec pages-left))
+                          (vec (concat acc users))
+                          retries-left
+                          (:scroll_param body))))
            (do (log/error "INTERCOM_REQUEST_FAILED" status body)
-               (Thread/sleep 3)
-               (recur pages-left
-                      acc
-                      scroll-param))))
-     acc))))
+               (if (pos? retries-left)
+                 (do
+                   (log/info "RETRYING" retries-left retries)
+                   (Thread/sleep 100)
+                   (recur pages-left
+                          acc
+                          (dec retries-left)
+                          scroll-param))
+                 (do
+                   (log/error "NO_MORE_RETRIES_LEFT" retries-left retries)
```

Comment:
The logs in swagger looks bad (generally), I suggest to pass `retries-left` and `retries` as a map, here and in all other places we print logs